extension/services/deezer/deezer.js

```diff
@@ -0,0 +1,80 @@
+import * as music from "../../intents/music/music.js";
+import * as serviceList from "../../background/serviceList.js";
+import * as content from "../../background/content.js";
+import { shouldDisplayWarning } from "../../limiter.js";
+
+class Deezer extends serviceList.Service {
+  async playQuery(query) {
+    await this.initTab("/services/deezer/player.js");
+    if(this.initTab){
+   window.alert ("Please signup or login to use this service. If you have, please ignore this message.");
```

Comment:
I think you can do:

```js
this.context.activateTab(this.tab);
const exc = new Error("Not logged in");
exc.displayMessage = "Please log in";
throw exc;
```

That said, this probably isn't the right place to raise this exception. The tab might get opened but the user might still be logged in. Instead you should let `player.js` raise the exception and catch it